tests/testthat/test-conjunct_expectation_linter.R

```diff
@@ -0,0 +1,38 @@
+test_that("conjunct_expectation_linter skips allowed usages", {
+  expect_lint("expect_true(x)", NULL, conjunct_expectation_linter())
+  expect_lint("testthat::expect_false(x, y, z)", NULL, conjunct_expectation_linter())
+
+  # more complicated expression
+  expect_lint("expect_true(x || (y && z))", NULL, conjunct_expectation_linter())
+  # the same by operator precedence, though not obvious a priori
+  expect_lint("expect_true(x || y && z)", NULL, conjunct_expectation_linter())
+  expect_lint("expect_true(x && y || z)", NULL, conjunct_expectation_linter())
+})
+
+test_that("conjunct_expectation_linter blocks && conditions with expect_true()", {
+  expect_lint(
+    "expect_true(x && y)",
+    rex::rex("Instead of expect_true(A && B), write multiple unit tests"),
+    conjunct_expectation_linter()
+  )
+
+  expect_lint(
+    "expect_true(x && y && z)",
+    rex::rex("Instead of expect_true(A && B), write multiple unit tests"),
+    conjunct_expectation_linter()
+  )
+})
+
+test_that("conjunct_expectation_linter blocks || conditions with expect_false()", {
+  expect_lint(
+    "expect_false(x || y)",
+    rex::rex("Instead of expect_false(A || B), write multiple unit tests"),
+    conjunct_expectation_linter()
+  )
+
+  expect_lint(
+    "expect_false(x || y || z)",
```

Comment:
Maybe additionally check `expect_false(x && y || z)`?

Reply:
no because `||` has higher precedence:

```
substitute(x && y || z)[[1]]
# `||`
```

Ditto for `x || y && z`.  It does make sense to have some better `expect_false()` false positive checks though.

Reply:
Exactly? So `expect_false(x && y || z)` is equivalent to `expect_false(x && y); expect_false(z)`.

Reply:
ah I see your point now